rxandroidble/src/main/java/com/polidea/rxandroidble2/RxBleDevice.java

```diff
@@ -109,6 +109,48 @@
      */
     Observable<RxBleConnection> establishConnection(boolean autoConnect, @NonNull Timeout operationTimeout);
 
+    /**
+     * Establishes connection with a given BLE device. {@link RxBleConnection} is a handle, used to process BLE operations with a connected
+     * device.
+     * <p>
+     * The connection is automatically disconnected (and released) when resulting Observable is unsubscribed.
+     * On the other hand when the connections is interrupted by the device or the system, the Observable will be unsubscribed as well
+     * following BleDisconnectedException or BleGattException emission.
+     * <p>
+     * During the disconnect process the library automatically handles order and requirement of device disconnect and gatt close operations.
+     * <p>
+     * Autoconnect concept may be misleading at first glance. In cases when the BLE device is available and it is advertising constantly you
+     * won't need to use autoconnect. Use autoconnect for connections where the BLE device is not advertising at
+     * the moment of #establishConnection call.
+     *
+     * @param autoConnect      Flag related to
+     *                         {@link android.bluetooth.BluetoothDevice#connectGatt(Context, boolean, BluetoothGattCallback)}
+     *                         autoConnect flag.
+     *                         If false, the connection will fail with {@link com.polidea.rxandroidble2.exceptions.BleGattException} if
+     *                         the device is not in range after a 30-second timeout. If true, the connection will be pending indefinitely.
+     *                         Unlike the native Android API, if set to true and the connection is lost there will NOT be any attempt to
+     *                         reconnect unless explicitly resubscribed.
+     * @param operationTimeout Timeout configuration after which the operation will be considered as broken. Eventually the operation
+     *                         will be canceled and removed from queue. Keep in mind that it will cancel the library's operation
+     *                         only and may leave Android's BLE stack in an inconsistent state.
+     * @param connectingTimeout   Timeout configuration after which the connection will be considered as broken. Eventually the connection
+     *                         attempt will be disconnected and then closed when the connection subscription is released.
```

Comment:
I would add here information that it is only taken onto account when used with `autoConnect=false`.